plugins/Transitions/API.php

```diff
@@ -81,25 +84,39 @@ public function getTransitionsForAction($actionName, $actionType, $idSite, $peri
             'date' => Period\Factory::build($period->getLabel(), $date)->getLocalizedShortString()
         );
 
-        $partsArray = explode(',', $parts);
-        if ($parts == 'all' || in_array('internalReferrers', $partsArray)) {
-            $this->addInternalReferrers($logAggregator, $report, $idaction, $actionType, $limitBeforeGrouping);
-        }
-        if ($parts == 'all' || in_array('followingActions', $partsArray)) {
-            $includeLoops = $parts != 'all' && !in_array('internalReferrers', $partsArray);
-            $this->addFollowingActions($logAggregator, $report, $idaction, $actionType, $limitBeforeGrouping, $includeLoops);
-        }
-        if ($parts == 'all' || in_array('externalReferrers', $partsArray)) {
-            $this->addExternalReferrers($logAggregator, $report, $idaction, $actionType, $limitBeforeGrouping);
-        }
+        try {
+            $partsArray = explode(',', $parts);
+            if ($parts == 'all' || in_array('internalReferrers', $partsArray)) {
+                $this->addInternalReferrers($logAggregator, $report, $idaction, $actionType, $limitBeforeGrouping);
+            }
+            if ($parts == 'all' || in_array('followingActions', $partsArray)) {
+                $includeLoops = $parts != 'all' && !in_array('internalReferrers', $partsArray);
+                $this->addFollowingActions($logAggregator, $report, $idaction, $actionType, $limitBeforeGrouping, $includeLoops);
+            }
+            if ($parts == 'all' || in_array('externalReferrers', $partsArray)) {
+                $this->addExternalReferrers($logAggregator, $report, $idaction, $actionType, $limitBeforeGrouping);
+            }
 
-        // derive the number of exits from the other metrics
-        if ($parts == 'all') {
-            $report['pageMetrics']['exits'] = $report['pageMetrics']['pageviews']
-                - $this->getTotalTransitionsToFollowingActions()
-                - $report['pageMetrics']['loops'];
+            // derive the number of exits from the other metrics
+            if ($parts == 'all') {
+                $report['pageMetrics']['exits'] = $report['pageMetrics']['pageviews']
+                    - $this->getTotalTransitionsToFollowingActions()
+                    - $report['pageMetrics']['loops'];
+            }
+        } catch (\Exception $e) {
+            Model::handleMaxExecutionTimeError(
```

Comment:
Is there a chance that if the query doesn't fail because of max execution error, the exception is not thrown?

Reply:
oh sure. there should be a `throw $e` after that. Will add...